core/arch/arm/mm/core_mmu.c

```diff
@@ -596,6 +596,8 @@ uint32_t core_mmu_type_to_attr(enum teecore_memtypes t)
 		return attr | TEE_MATTR_SECURE | TEE_MATTR_PR | cached;
 	case MEM_AREA_TEE_RAM_RW:
 		return attr | TEE_MATTR_SECURE | TEE_MATTR_PRW | cached;
+	case MEM_AREA_TEE_COHERENT:
+		return attr | TEE_MATTR_SECURE | TEE_MATTR_PRWX | noncache;
```

Comment:
I need a piece of onchip RAM for low power suspend/resume and idle features. This piece memory is about 32KB which contains data and code, I could only use section mapping here, because page mapping will consume more memory for page tables. I need this piece memory to be non-cacheable which could ease the operation. I also thought about split W and X to different page, but not easy to implement.